ractor_core.h

```diff
@@ -1,3 +1,4 @@
+#include "internal/gc.h"
```

Comment:
because `ractor_core.h` is relying on some functions defined inside `internal/gc.h` which were previously included implicitly through the inclusion of `vm_core.h`. Now that `vm_core.h` doesn't include `internal/gc.h` anymore we need to specify this dependancy explicitly.